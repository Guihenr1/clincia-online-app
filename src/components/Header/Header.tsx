import { FC, MouseEvent, useState } from "react";
import useStyles from "./styles";
import { AppBar, Box, IconButton, Tab, Tabs, Toolbar } from "@mui/material";
import LogoutIcon from "@mui/icons-material/Logout";

interface Menu {
  text: string;
  onClick: (event: any) => void;
}

interface HeaderProps {
  menu: Menu[];
  handleClickLogout: Function;
}

const Header: FC<HeaderProps> = ({ menu, handleClickLogout }) => {
  const s = useStyles();
  const [tab, setTab] = useState(0);

  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setTab(newValue);
  };

  return (
    <AppBar>
      <Toolbar>
        <Box marginTop={2} marginBottom={2} className={s.classes.header}>
          <img src="logo.png" alt="logo" />
          <Tabs
            className={s.classes.margin}
            value={tab}
            textColor="inherit"
            onChange={handleChange}
          >
            {menu.map((item, index) => (
              <Tab
                label={item.text}
                key={index}
                onClick={(e) => item.onClick(e)}
              />
            ))}
          </Tabs>
          <IconButton
            id="logout"
            color="inherit"
            onClick={() => handleClickLogout()}
          >
            <LogoutIcon />
          </IconButton>
        </Box>
      </Toolbar>
    </AppBar>
  );
};

Header.defaultProps = {};

export default Header;
