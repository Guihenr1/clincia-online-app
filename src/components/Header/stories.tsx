import { Meta } from "@storybook/react";
import Header from "./Header";

export default {
  component: Header,
  args: {
    children: "",
    handleClickLogout: () => console.log("Logout"),
    menu: [
      {
        text: "Home",
        onClick: () => console.log("Home"),
      },
      {
        text: "Privacy",
        onClick: () => console.log("Privacy"),
      },
      {
        text: "Settings",
        onClick: () => console.log("Settings"),
      },
    ],
  },
} as Meta;

export const Primary = (args: any) => <Header {...args} />;
