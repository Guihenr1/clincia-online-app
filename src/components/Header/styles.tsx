import { makeStyles } from "tss-react/mui";

const useStyles = makeStyles()(() => ({
  header: {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    width: "100%",
  },
  margin: {
    marginLeft: "auto",
  },
}));

export default useStyles;
