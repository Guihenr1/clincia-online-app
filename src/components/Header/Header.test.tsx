import { fireEvent, render, screen } from "@testing-library/react";
import Header from "./Header";
import "@testing-library/jest-dom/extend-expect";

describe("Header", () => {
  const tab1: string = "Home";
  const tab2: string = "Privacy";
  const onClickMockTab1 = jest.fn();
  const onClickMockTab2 = jest.fn();
  const menu = [
    {
      text: tab1,
      onClick: () => onClickMockTab1(),
    },
    {
      text: tab2,
      onClick: () => onClickMockTab2(),
    },
  ];

  const handleClickLogout = jest.fn();

  test("renders the Header", () => {
    const { container } = render(
      <Header menu={menu} handleClickLogout={handleClickLogout} />
    );

    expect(screen.getByText(tab1)).toBeInTheDocument();
    expect(screen.getByText(tab1)).toHaveClass("Mui-selected");
    expect(screen.getByText(tab2)).not.toHaveClass("Mui-selected");

    expect(container).toMatchSnapshot();
  });

  test("call onClick", () => {
    render(<Header menu={menu} handleClickLogout={handleClickLogout} />);

    expect(screen.getByText(tab1)).toBeInTheDocument();
    expect(screen.getByText(tab1)).toHaveClass("Mui-selected");
    expect(screen.getByText(tab2)).not.toHaveClass("Mui-selected");

    fireEvent.click(screen.getByText(tab2));
    expect(screen.getByText(tab1)).not.toHaveClass("Mui-selected");
    expect(screen.getByText(tab2)).toHaveClass("Mui-selected");
    expect(onClickMockTab2).toHaveBeenCalledTimes(1);

    fireEvent.click(screen.getByText(tab1));
    expect(screen.getByText(tab1)).toHaveClass("Mui-selected");
    expect(screen.getByText(tab2)).not.toHaveClass("Mui-selected");
    expect(onClickMockTab1).toHaveBeenCalledTimes(1);
  });

  test("call logout", () => {
    render(<Header menu={menu} handleClickLogout={handleClickLogout} />);

    fireEvent.click(screen.getByRole("button"));
    expect(handleClickLogout).toHaveBeenCalledTimes(1);
  });
});
