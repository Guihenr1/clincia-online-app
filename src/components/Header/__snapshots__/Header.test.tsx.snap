// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the Header 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        class="MuiBox-root css-1ai4e04-header"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <div
          class="MuiTabs-root css-e0p3et-MuiTabs-root-margin"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-bottom: 0px;"
          >
            <div
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected css-cjctlb-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="0"
                type="button"
              >
                Home
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-cjctlb-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Privacy
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          id="logout"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LogoutIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
